radixdlt-core/radixdlt/src/test/java/com/radixdlt/network/messaging/MessagePreprocessorTest.java

```diff
@@ -0,0 +1,49 @@
+package com.radixdlt.network.messaging;
+
+import com.radixdlt.DefaultSerialization;
+import com.radixdlt.counters.SystemCountersImpl;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.network.p2p.NodeId;
+import com.radixdlt.network.p2p.PeerControl;
+import org.junit.Test;
+
+import java.time.Duration;
+
+import static org.junit.Assert.assertFalse;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class MessagePreprocessorTest {
+    // serialized version of SyncRequestMessage(null),
+    // can't be created with code, as SyncRequestMessage does not accept null anymore
+    private static final byte[] SYNC_REQUEST_MESSAGE = {
```

Comment:
Does it need to be hardcoded?

Reply:
No, if we find a convenient way to generate incorrect message on demand.

Reply:
I did mean convenient way.

Reply:
done